src/game_display.cpp

```diff
@@ -647,18 +654,31 @@ std::vector<texture> game_display::get_reachmap_images(const map_location& loc)
 	std::vector<std::string> names;
 	const auto adjacent = get_adjacent_tiles(loc);
 
-	enum visibility { REACH = 0, CLEAR = 1 };
+	enum visibility { REACH = 0, ENEMY = 1, CLEAR = 2 };
 	std::array<visibility, 6> tiles;
 
 	const std::string* image_prefix_ = &game_config::reach_map_prefix;
 	DBG_DP << "Loaded image prefix: " << game_config::reach_map_prefix;
 
+	// Get the reachmap-context team, if no unit is selected, the reachmap belongs to the displayed unit
```

Comment:
These lines handling the reachmap-context team causes a crash, working on it right now.

The explanation is because this assumes that a unit is selected or displayed when the reachmap is being drawn, which isn't always the case (a unit may die while its reachmap is being displayed, causing a crash).

Reply:
Also the reachmap-context team doesn't really need to be recalculated every time a hex is drawn (as is now), but every time the reachmap is drawn. Probably should move it to its own function and call it in `void display::process_reachmap_changes()` from display.cpp